script/cargo.py

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env python
+
+import argparse
+import os
+import sys
+import subprocess
+import shutil
+
+
+def main():
+    args = parse_args()
+
+    is_debug = args.is_debug[0]
+    manifest_dir = args.manifest_dir[0]
+    output = args.output[0]
+
+    build(is_debug, manifest_dir, output)
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(description='Cargo')
+
+    parser.add_argument('--is_debug', nargs=1)
+    parser.add_argument('--manifest_dir', nargs=1)
+    parser.add_argument('--output', nargs=1)
+
+    args = parser.parse_args()
+
+    # validate args
+    if (args.is_debug is None or
+        len(args.is_debug) is not 1 or
+            len(args.is_debug[0]) is 0):
+        raise Exception("is_debug argument was not specified correctly")
+
+    print args.is_debug[0]
+    if (args.is_debug[0] != "false" and args.is_debug[0] != "true"):
+        raise Exception("is_debug argument was not specified correctly")
+
+    if (args.manifest_dir is None or
+        len(args.manifest_dir) is not 1 or
+            len(args.manifest_dir[0]) is 0):
+        raise Exception("manifest_dir argument was not specified correctly")
+
+    if (args.output is None or
+        len(args.output) is not 1 or
+            len(args.output[0]) is 0):
+        raise Exception("output argument was not specified correctly")
+
+    if "out" not in args.output[0]:
+        raise Exception("output did not contain 'out'")
+
+    # args are valid
+    return args
+
+
+def get_platform():
+    platforms = {
+        'win32' : 'Windows',
+        'darwin' : 'macOS',
+        'linux1' : 'Linux',
+        'linux2' : 'Linux'
+    }
+
+    if sys.platform not in platforms:
+        return sys.platform
+
+    return platforms[sys.platform]
+
+
+def build(is_debug, manifest_dir, output):
+    platform = get_platform()
+    if platform == "Windows":
+        target = "x86_64-pc-windows-msvc"
+    elif platform == "macOS":
+        target = "x86_64-apple-darwin"
+    elif platform == "Linux":
+        target = "x86_64-unknown-linux-musl"
+
+    args = []
+    args.append("cargo")
+    args.append("build")
+    if is_debug == "false":
+        args.append("--release")
+    args.append("--manifest-path=" + manifest_dir)
+    args.append("--target-dir=" + output)
+    args.append("--target=" + target)
+
+    env = os.environ
+    env["PATH"] = "NO_CXXEXCEPTIONS" + env["PATH"]
```

Comment:
we aren't adding this to the path, it should be `env['NO_CXXEXCEPTIONS'] = 1`

Reply:
👍